### PR TITLE
update: forbid updating repo-imports with --to-remote

### PR DESCRIPTION
**Pull request by isidentical**
* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here. (will mention in iterative/dvc.org/pull/2175)

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

**Review by efiop — APPROVED** [review]

**Comment by jorgeorpinel** [comment]
Hi @isidentical I see there's a docs PR for this (https://github.com/iterative/dvc.org/pull/2175/files), thanks!

I'm just not sure this was mentioned completely? What was forbidden exactly? It seems like update works perfectly well with --to-remote imports. Maybe I just don't understand this PR.

File: dvc/stage/imports.py [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
+from dvc.exceptions import InvalidArgumentError
+
 logger = logging.getLogger(__name__)
 
 
 def _update_import_on_remote(stage, remote, jobs):
+    if stage.is_repo_import:
+        raise InvalidArgumentError(
+            "Can't update a repo import with --to-remote"
+        )
```
Comment (jorgeorpinel) [inline dvc/stage/imports.py:12]: What's a repo import?

**Comment by jorgeorpinel** [comment]
Oh I see "repo import" means a `dvc import` (not `import-url`). OK please see https://github.com/iterative/dvc.org/pull/2175#pullrequestreview-596702408

**Comment by jorgeorpinel** [comment]
BTW I'm not sure that term will mean anything for users. If that exception is a user message I would rephrase to something more explicit like "Data imported with `dvc import` can't be updated --to-remote." or "Only data imported with `dvc import-url --to-remote` can be updated `--to-remote`."

File: dvc/stage/imports.py (hunk 1 above)
Comment (jorgeorpinel) [reply dvc/stage/imports.py:12]: BTW I'm not sure that term will mean anything for users. If that exception is a user message I would rephrase to something more explicit like "Data imported with `dvc import` can't be updated `--to-remote`." or "Only data imported with `dvc import-url --to-remote` can be updated `--to-remote`."

**Comment by isidentical** [comment]
> Oh I see "repo import" means a dvc import (not import-url). OK please see iterative/dvc.org#2175 (review)

I agree that the error message is a bit term-focused, even though it is the correct term. Let's just use `Only data imported with dvc import-url can be updated with --to-remote.`

**Comment by jorgeorpinel** [comment]
Yes or maybe more general like "Data imported from other DVC or Git repositories can't be updated to-remote." to avoid terminology completely. Although maybe mentioning the exact command is better. Up to you

**Comment by jorgeorpinel** [comment]
p.s. is there a core issue to put in these ideas?

**Comment by isidentical** [comment]
> p.s. is there a core issue to keep track of these ideas?

nop, though feel free to just create an issue and CC me
